recording vs sampled explained (#291)
open-telemetry/opentelemetry-specification: specification/sdk-tracing.md

All other sampling algorithms may be implemented on SDK layer in exporters, or
even out of process in Agent or Collector.

> OpenTelemetry API has two properties responsible for the data collection:

Reviewer: ```suggestion
The OpenTelemetry API has two properties responsible for the data collection:
```

> * `IsRecordEvents` field of a `Span`

Reviewer 2: IsRecord**ing**Events

> * `IsRecordEvents` field of a `Span`

Reviewer: ```suggestion
* `IsRecording` field of a `Span`
```

>   * If `true` the current `Span` records tracing events (attributes, events,

Reviewer: I wouldn't make this a sublist of bullet points but rather just a single paragaph at the top-level point.

    status, etc.), otherwise all tracing events are dropped.
  * Users can use this property to determine if expensive trace events can be
    avoided.

>   * [Span Processor](#span-processor) will receive all spans with this flag set.

Reviewer: ```suggestion
  * [Span Processors](#span-processor) will receive all spans with this flag set.
```

    However, [Span Exporter](#span-exporter) will not receive them unless the
    `Sampled` flag was set.

> * `Sampled` `TraceFlag` on `SpanContext`

Reviewer: ```suggestion
* `Sampled` flag in `TraceFlags` on `SpanContext`
```

>   * This flag is propagated via the `TraceFlag` to the child Spans. For more

Reviewer: ```suggestion
  * This flag is propagated via the `SpanContext` to child Spans. For more
```

>   * This flag is propagated via the `TraceFlag` to the child Spans. For more

Reviewer: Same as above: I wouldn't make this a sublist of bullet points but rather just a single paragaph at the top-level point.

>     details see the w3c definition [here][trace-flags].

Reviewer: ```suggestion
    details see the [W3C specification][trace-flags].
```

>   * This flag indicates that `Span` is being `sampled` and exported.

Reviewer 2: the/a Span

>   * This flag indicates that `Span` is being `sampled` and exported.

Reviewer 2: the/a Span

>   * This flag indicates that `Span` is being `sampled` and exported.

Reviewer: ```suggestion
  * This flag indicates that the `Span` has been `sampled` and will be exported.
```

  * [Span Processor](#span-processor) and [Span Exporter](#span-exporter) will
    receive spans with the `Sampled` flag set for processing.

> The flag combination `SampledFlag == false` and `IsRecordEvents == true` means

Reviewer 2: Either `RecordEvents` or `IsRecordingEvents`?

> The flag combination `SampledFlag == false` and `IsRecordEvents == true` means

Reviewer: ```suggestion
The flag combination `SampledFlag == false` and `IsRecording == true` means
```

that the current `Span` does record information, but most likely the child
`Span` will not.

> The flag combination `SampledFlag == true` and `RecordEvents == false` can cause

Reviewer 2: Either `RecordEvents` or `IsRecordingEvents`?

> The flag combination `SampledFlag == true` and `RecordEvents == false` can cause

Reviewer: ```suggestion
The flag combination `SampledFlag == true` and `RecordEvents == false` could cause
```

gaps in the distributed trace, and because of this OpenTelemetry API MUST NOT
allow this combination.

> SDK defines two interfaces - [`Sampler`](#sampler) and [`Decision`](#decision)

Reviewer: ```suggestion
The SDK defines the two interfaces [`Sampler`](#sampler) and [`Decision`](#decision)
```

All other sampling algorithms may be implemented on SDK layer in exporters, or
even out of process in Agent or Collector.

The OpenTelemetry API has two properties responsible for the data collection:

> * `IsRecordingEvents` field of a `Span`. If `true` the current `Span` records

Reviewer 3: ```suggestion
* `IsRecording` field of a `Span`. If `true` the current `Span` records
```

  tracing events (attributes, events, status, etc.), otherwise all tracing
  events are dropped. Users can use this property to determine if expensive
  trace events can be avoided. [Span Processors](#span-processor) will receive
  all spans with this flag set. However, [Span Exporter](#span-exporter) will
  not receive them unless the `Sampled` flag was set.
* `Sampled` flag in `TraceFlags` on `SpanContext`. This flag is propagated via
  the `SpanContext` to child Spans. For more details see the [W3C
  specification][trace-flags]. This flag indicates that the `Span` has been
  `sampled` and will be exported. [Span Processor](#span-processor) and [Span
  Exporter](#span-exporter) will receive spans with the `Sampled` flag set for
  processing.

> The flag combination `SampledFlag == false` and `IsRecordingEvents == true`

Reviewer 3: ```suggestion
The flag combination `SampledFlag == false` and `IsRecording == true`
```

means that the current `Span` does record information, but most likely the child
`Span` will not.

> The flag combination `SampledFlag == true` and `IsRecordingEvents == false`

Reviewer 3: ```suggestion
The flag combination `SampledFlag == true` and `IsRecording == false`
```
